### PR TITLE
ci: upgrade to golang 1.18

### PR DESCRIPTION
**Pull request by cgorenflo**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,20 +22,22 @@ lint:
...
-	go get -u github.com/matryer/moq
-endif
-ifeq (which mdformat,)
-	pip3 install mdformat
-endif
-ifeq (which protoc,)
-	@echo "Please install protoc for grpc (https://grpc.io/docs/languages/go/quickstart/)"
-endif
+	@which goimports &>/dev/null	||	go install golang.org/x/tools/cmd/goimports
+	@which moq &>/dev/null			||	go install github.com/matryer/moq
```
Comment (milapsheth) [inline Makefile:38]:
```suggestion
	@which goimports &>/dev/null	||	go install golang.org/x/tools/cmd/goimports@latest
	@which moq &>/dev/null			||	go install github.com/matryer/moq@latest
```
Perhaps?

File: Makefile [hunk 2]
```diff
@@ -22,20 +22,22 @@ lint:
...
+	@which mdformat &>/dev/null 	||	pip3 install mdformat
+	@which protoc &>/dev/null 		|| 	echo "Please install protoc for grpc (https://grpc.io/docs/languages/go/quickstart/)"
```
Comment (milapsheth) [inline Makefile:40]: `protoc` isn't needed for this repo I believe

File: Makefile [hunk 3]
```diff
@@ -22,20 +22,22 @@ lint:
...
-	go generate -x ./...
+generate: go.sum prereqs goimports
+	go generate ./...
 
-.PHONY: prereqs
+.PHONY: goimports
+goimports:
+	@echo "running goimports"
+# exclude mocks and proto generated files
+	@goimports -l -local github.com/axelarnetwork/ . | grep -v .pb.go$ | grep -v mock | xargs goimports -local github.com/axelarnetwork/ -w
```
Comment (milapsheth) [inline Makefile:32]:
```suggestion
	@goimports -l -local github.com/axelarnetwork/ . | grep -v mock | xargs goimports -local github.com/axelarnetwork/ -w
```

File: go.mod [hunk 4]
```diff
@@ -11,8 +11,80 @@ require (
 	github.com/spf13/cobra v1.2.1
 	github.com/stretchr/testify v1.7.0
 	github.com/tendermint/tendermint v0.34.14
```
Comment (milapsheth) [inline go.mod:13]:
```suggestion
	github.com/tendermint/tendermint v0.34.16
```

File: go.mod [hunk 5]
```diff
@@ -11,8 +11,80 @@ require (
 	github.com/spf13/cobra v1.2.1
 	github.com/stretchr/testify v1.7.0
```
Comment (milapsheth) [inline go.mod:12]: Update all the above deps too. utils had a new commit recently. You can copy paste from [here](https://github.com/axelarnetwork/tm-events/pull/38/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R6)

File: Makefile (hunk 1 above)
Comment (cgorenflo) [reply Makefile:38]: no, it's supposed to grab the one specified in the go.mod file

**Review by milapsheth — APPROVED** [review]

**Pull request merged**